app/lib/three_scale/spam_protection/checks/base.rb

```diff
@@ -1,31 +1,49 @@
+# frozen_string_literal: true
+
 module ThreeScale::SpamProtection
   module Checks
 
+    HIDE_STYLE = "position: absolute; width: 1px; height: 0; left: 0; overflow: hidden;"
+    SPAM_CHECKS_SECRET_KEY = -> { Rails.application.key_generator.generate_key('spam-protection-checks', 32) }
+
     class Base
-      attr_reader :config
+      def initialize(store)
+        @store = store
+      end
 
-      def initialize(config)
-        @config = config
+      def input(_form)
+        ""
       end
 
       def name
         self.class.to_s.underscore
       end
 
-      def apply!(klass)
-        # do nothing
+      private
+
+      def encryptor
+        @encryptor ||= ActiveSupport::MessageEncryptor.new(SPAM_CHECKS_SECRET_KEY.call)
```

Comment:
We will have different encryptors for different workers, correct? How does this affect the logic?